File: easy/2379_minimum_recolors_to_get_k_consecutive_black_blocks/solution.cpp
#include <string>

class Solution {
public:
  int minimumRecolors(std::string blocks, int k) {
    int whiteCount = 0;
    int blackCount = 0;

    for (size_t i = 0; i < k; ++i) {
      if (blocks[i] == 'W')
        ++whiteCount;
      else
        ++blackCount;
    }

    int recolorsCount = k - blackCount;

    if (blocks.size() == k)
      return recolorsCount;

    for (size_t i = k; i < blocks.size(); ++i) {
      if (blocks[i - k] == 'W')
        --whiteCount;
      else
        --blackCount;

      if (blocks[i] == 'W')
        ++whiteCount;
      else
        ++blackCount;

      recolorsCount = std::min(recolorsCount, k - blackCount);
    }

    return recolorsCount;
  }
};

File: easy/2379_minimum_recolors_to_get_k_consecutive_black_blocks/solution.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        black_count = 0

        for i in range(k):
            if blocks[i] == "B":
                black_count += 1

        recolors_count = k - black_count

        if len(blocks) == k:
            return recolors_count

        for i in range(k, len(blocks)):
            if blocks[i - k] == "B":
                black_count -= 1

            if blocks[i] == "B":
                black_count += 1

            recolors_count = min(recolors_count, k - black_count)

        return recolors_count
